src/tools/illink/test/Mono.Linker.Tests.Cases/RequiresCapability/RequiresOnAttribute.cs

```diff
@@ -24,7 +29,15 @@ public static void Main ()
 			_fieldWithAttributeWhichRequires = 0;
 			PropertyWithAttributeWhichRequires = false;
 			TestMethodWhichRequiresWithAttributeWhichRequires ();
+			RequiresTriggeredByAttributeUsage.Test ();
+
+			// Accessing the members to test via direct calls/access is not enough
+			// in NativeAOT custom attributes are only looked at if the member is
+			// reflection enabled - so there has to be a reflection access to it somewhere.
+			// On the other hand the analyzer reports RDC/RAF only on direct access. So we need to have "both"
```

Comment:
This goes to the larger discussion on how should linker treat attributes. Currently we're very conservative in that we keep all instantiations pretty much everywhere. NativeAOT is also conservative, but less so, it will keep all attributes but only on reflectable entities.

I do agree that it would make sense to change linker to behave more like NativeAOT - which is aligned with the long term goal of linker optimizing non-reflectable entities more. Currently we only remove parameter names for non-reflectable methods, we already have issues and ides for example to remove field names for non-reflectable fields and so on. Removing attributes on these would also fall into that bucket.

Ideally we would be able to remove unused attributes, which means the same behavior as NativeAOT, but also only keep attributes which are actually asked for. But that is pretty hard to track correctly - basically `GetCustomAttributes` API would have to be RUC and intrinsic. That would probably break lot of things.